lib/MultiCompiler.js

```diff
@@ -2,175 +2,176 @@
-					commonPath = commonPath.replace(/[\/\\][^\/\\]*$/, "");
+			get: () => {
+				let commonPath = compilers[0].outputPath;
+				for(const compiler of compilers) {
+					while(compiler.outputPath.indexOf(commonPath) !== 0 && /[\/\\]/.test(commonPath)) {
+						commonPath = commonPath.replace(/[\/\\][^\/\\]*$/, "");
+					}
 				}
-			}
-			if(!commonPath && compilers[0].outputPath[0] === "/") return "/";
-			return commonPath;
-		}
-	});
 
-	var doneCompilers = 0;
-	var compilerStats = [];
-	this.compilers.forEach(function(compiler, idx) {
-		var compilerDone = false;
-		compiler.plugin("done", function(stats) {
-			if(!compilerDone) {
-				compilerDone = true;
-				doneCompilers++;
-			}
-			compilerStats[idx] = stats;
-			if(doneCompilers === this.compilers.length) {
-				this.applyPlugins("done", new MultiStats(compilerStats));
-			}
-		}.bind(this));
-		compiler.plugin("invalid", function() {
-			if(compilerDone) {
-				compilerDone = false;
-				doneCompilers--;
+				if(!commonPath && compilers[0].outputPath[0] === "/") return "/";
+				return commonPath;
 			}
-			this.applyPlugins("invalid");
-		}.bind(this));
-	}, this);
-}
-module.exports = MultiCompiler;
+		});
 
-MultiCompiler.prototype = Object.create(Tapable.prototype);
-MultiCompiler.prototype.constructor = MultiCompiler;
+		let doneCompilers = 0;
+		let compilerStats = [];
+		this.compilers.forEach((compiler, idx) => {
+			let compilerDone = false;
+			compiler.plugin("done", stats => {
+				if(!compilerDone) {
+					compilerDone = true;
+					doneCompilers++;
+				}
+				compilerStats[idx] = stats;
+				if(doneCompilers === this.compilers.length) {
+					this.applyPlugins("done", new MultiStats(compilerStats));
+				}
+			});
+			compiler.plugin("invalid", () => {
+				if(compilerDone) {
+					compilerDone = false;
+					doneCompilers--;
+				}
+				this.applyPlugins("invalid");
+			});
+		}, this);
+	}
 
-function runWithDependencies(compilers, fn, callback) {
-	var fulfilledNames = {};
-	var remainingCompilers = compilers;
+	get inputFileSystem() {
+		return {
+			configurable: false,
+			get: (() => {
+				throw new Error("Cannot read inputFileSystem of a MultiCompiler");
+			})()
+		};
+	}
 
-	function isDependencyFulfilled(d) {
-		return fulfilledNames[d];
+	get outputFileSystem() {
+		return {
+			configurable: false,
+			get: (() => {
+				throw new Error("Cannot read outputFileSystem of a MultiCompiler");
+			})()
+		};
 	}
 
-	function getReadyCompilers() {
-		var readyCompilers = [];
-		var list = remainingCompilers;
-		remainingCompilers = [];
-		for(var i = 0; i < list.length; i++) {
-			var c = list[i];
-			var ready = !c.dependencies || c.dependencies.every(isDependencyFulfilled);
-			if(ready)
-				readyCompilers.push(c);
-			else
-				remainingCompilers.push(c);
-		}
-		return readyCompilers;
+	set inputFileSystem(value) {
+		this.compilers.forEach(compiler => {
+			compiler["inputFileSystem"] = value;
```

Comment:
Changed setters, inputFileSystem and outputFileSystem to use dot notation instead of brackets